compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensionsOps.td

```diff
@@ -368,4 +368,40 @@ def ConfigExtractPart :
   }];
 }
 
+def ApplyBufferOptimizationsOp :
+  Op<Transform_Dialect, "iree.apply_buffer_optimizations",
+    [TransformEachOpTrait,
+     TransformOpInterface,
+     DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
+  let description = [{
+    This applies memory optimization on memref. In particular it does store to
+    load forwarding, dead store elimination and dead alloc elimination.
+
+    #### Return modes
+
+    This operation apples a set of memory optimization on the whole region of
+    the operand.
```

Comment:
in practice the impl seems to require a FuncOp, please specialize to that op.

Reply:
Why does it require a FuncOp? It should work on any region as far as I know.

Reply:
Ah my bad .. I did a code search and saw: ```static void eraseDeadAllocAndStores(func::FuncOp funcOp) {```

But that is in another location, feel free to disregard this and the specific ` def Transform_FuncOp : Transform_ConcreteOpType<"func.func">;` comment ..